rainbow-me/rainbow
# Tophat

## What changed (plus any additional context for devs)

Splits CI jobs in two parallel.

- build: which generates a ready to install sim and iPhone build for the current comment in release mode.
- tests: which runs linting + unit tests in one action, and e2e in another one (also in parallel)


## Screen recordings / screenshots

Sim: 

https://github.com/user-attachments/assets/170b203a-699f-4e1d-8274-e4aea983ceb5

Device:



https://github.com/user-attachments/assets/e431624e-83b2-40b7-9432-82dcfba44889



## What to test

- Install the latest version of tophat.
- Make sure you have the  iPhone 15 pro simulator with iOS 17.4 installed.
- Make sure you have your phone as a device registered inside xcode (with wifi builds enabled)
- Select your iPhone on tophat and click on the comment of this PR where it says "device". The app should be installed immediately
- Select your Sim on tophat and click on the comment of this PR where it says "simulator". The app should be installed immediately
